### PR TITLE
[DOCS-622] Create dedicated prometheus guide for each context.

### PR DESCRIPTION
**Pull request by l0k0ms**
### What does this PR do?

Breaks down the GS into full on guide that are focus on the context in which the migration happens.

The idea is to reference those guides in the prometheus/OM integration page, or from other places of the documentation.

### Preview link

* [Host collection](https://docs-staging.datadoghq.com/gus/prometheus/integrations/guide/prometheus-host-collection/)
* [Docker collection](https://docs-staging.datadoghq.com/gus/prometheus/integrations/guide/prometheus-docker-collection/?tab=standard)
* [Kubernetes collection](https://docs-staging.datadoghq.com/gus/prometheus/integrations/guide/prometheus-kubernetes-collection/)

### CODE REVIEW

File: content/en/agent/docker/prometheus.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+Starting with version 6.5.0, the Agent includes [OpenMetrics][3] and [Prometheus][4] checks capable of scraping Prometheus endpoints. Datadog recommends using the OpenMetrics check since it is more efficient and fully supports Prometheus text format. For more advanced usage of the `OpenMetricsCheck` interface, including writing a custom check, see the [Developer Tools][5] section. Use the Prometheus check only when the metrics endpoint does not support a text format.
+
+This page explains the basic usage of these checks, enabling you to import all your prometheus exposed metrics within Datadog.
+
+## Setup
+
+### Installation
+
+Launch the Docker Agent next to you other containers by replacing `<DATADOG_API_KEY>` with the API key for your organisation in the command below:
```
Comment (sarina-dd) [inline content/en/agent/docker/prometheus.md]:
```suggestion
Launch the Docker Agent next to your other containers by replacing `<DATADOG_API_KEY>` with the API key for your organization in the command below:
```

File: content/en/agent/docker/prometheus.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+**Note**: add `DD_SITE=datadoghq.eu` if you are using Datadog EU site.
+
+### Configuration
+
+The Agent detects if it's running on Docker and automatically searches all containers labels for Datadog-OpenMetrics labels. Autodiscovery expects labels to look like these examples, depending on the file type:
```
Comment (sarina-dd) [inline content/en/agent/docker/prometheus.md]:
```suggestion
The Agent detects if it's running on Docker and automatically searches all container labels for Datadog-OpenMetrics labels. Autodiscovery expects labels to look like these examples, depending on the file type:
```

File: content/en/agent/docker/prometheus.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+      text: 'Write your own custom Prometheus Check'
+---
+
+Collect your exposed Prometheus and OpenMetrics metrics from your application running inside your containers by using the Datadog Agent, and the [Datadog-OpenMetrics][1] or [Datadog-Prometheus][2] integrations.
+
+## Overview
+
+Starting with version 6.5.0, the Agent includes [OpenMetrics][3] and [Prometheus][4] checks capable of scraping Prometheus endpoints. Datadog recommends using the OpenMetrics check since it is more efficient and fully supports Prometheus text format. For more advanced usage of the `OpenMetricsCheck` interface, including writing a custom check, see the [Developer Tools][5] section. Use the Prometheus check only when the metrics endpoint does not support a text format.
+
+This page explains the basic usage of these checks, enabling you to import all your prometheus exposed metrics within Datadog.
```
Comment (cswatt) [inline content/en/agent/docker/prometheus.md]:
```suggestion
This page explains the basic usage of these checks, enabling you to import all your Prometheus exposed metrics within Datadog.
```

File: content/en/agent/docker/prometheus.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+| `<METRIC_TO_FETCH>`                      | Prometheus metrics key to be fetched from the Prometheus endpoint.                                                                                                                                             |
+| `<NEW_METRIC_NAME>`                      | Optional parameter which, if set, transforms the `<METRIC_TO_FETCH>` metric key to `<NEW_METRIC_NAME>` in Datadog. If you choose not to use this option, pass a list of strings rather than `key:value` pairs. |
+
+**Note**: See the [sample openmetrics.d/conf.yaml][6] for all available configuration options.
+
+## Getting started
+
+### Simple metric collection
+
+Here is a simple getting started to collect metrics exposed by your Prometheus running within a contrainer:
```
Comment (cswatt) [inline content/en/agent/docker/prometheus.md]:
```suggestion
To get started with collecting metrics exposed by Prometheus running within a container, follow these steps:
```

File: content/en/agent/docker/prometheus.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+    ```shell
+    DOCKER_CONTENT_TRUST=1 \
+    docker run -d -v /var/run/docker.sock:/var/run/docker.sock:ro \
+                  -v /proc/:/host/proc/:ro \
+                  -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
+                  -e DD_API_KEY="<DATADOG_API_KEY>" \
+                  datadog/agent:latest
+    ```
+
+2. To launch Prometheus in a container, the command [from the prometheus documentation][7] is: `docker run -p 9090:9090 prom/prometheus`. In order to notify the Agent to query this container with the openmetrics check we need to add the following configuration:
```
Comment (cswatt) [inline content/en/agent/docker/prometheus.md]:
```suggestion
2. To launch Prometheus in a container, run: `docker run -p 9090:9090 prom/prometheus`. In order to notify the Agent to query this container with the OpenMetrics check, add the following configuration:
```

i dont think we need to cite our sources

File: content/en/agent/docker/prometheus.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+    ```shell
+    -l com.datadoghq.ad.check_names='["openmetrics"]' -l com.datadoghq.ad.init_configs='[{}]' -l com.datadoghq.ad.instances='[  {"prometheus_url":"http://%%host%%:%%port%%/metrics","namespace":"documentation_example_docker","metrics":[ {"promhttp_metric_handler_requests_total": "prometheus.handler.requests.total"}]}]'
+    ```
+
+     To launch the prometheus container with the right annotations for autodiscovery to work run:
```
Comment (cswatt) [inline content/en/agent/docker/prometheus.md]:
```suggestion
     To launch the Prometheus container with the right annotations for Autodiscovery to work, run:
```

File: content/en/agent/kubernetes/prometheus.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+**Note**: See the [sample openmetrics.d/conf.yaml][8] for all available configuration options.
+
+## Getting Started
+
+### Simple metric collection
+
+1. [Launch the Datadog Agent][9].
+
+2. Use [this Prometheus Daemonset `prometheus.yaml`][10] to launch a Prometheus pod with already the autodiscovery configuration in it:
```
Comment (cswatt) [inline content/en/agent/kubernetes/prometheus.md]:
```suggestion
2. Use [this Prometheus DaemonSet `prometheus.yaml`][10] to launch a Prometheus pod with already the Autodiscovery configuration in it:
```

File: content/en/agent/kubernetes/prometheus.md [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+                  {
+                    "prometheus_url": "http://%%host%%:%%port%%/metrics",
+                    "namespace": "documentation_example_kubernetes",
+                    "metrics": [ {"promhttp_metric_handler_requests_total": "prometheus.handler.requests.total"}]
+                  }
+                ]
+      # (...)
+    ```
+
+     Command to create the prometheus pod:
```
Comment (cswatt) [inline content/en/agent/kubernetes/prometheus.md]:
```suggestion
     Command to create the Prometheus pod:
```

File: content/en/integrations/guide/prometheus-host-collection.md [hunk 9]
```diff
@@ -1,34 +1,28 @@
...
+Starting with version 6.5.0, the Agent includes [OpenMetrics][3] and [Prometheus][4] checks capable of scraping Prometheus endpoints. Datadog recommends using the OpenMetrics check since it is more efficient and fully supports Prometheus text format. For more advanced usage of the `OpenMetricsCheck` interface, including writing a custom check, see the [Developer Tools][5] section. Use the Prometheus check only when the metrics endpoint does not support a text format.
 
-This page explains the basic usage of these checks, which enable you to scrape custom metrics from Prometheus endpoints.
+This page explains the basic usage of these checks, enabling you to import all your prometheus exposed metrics within Datadog.
```
Comment (cswatt) [inline content/en/integrations/guide/prometheus-host-collection.md]:
```suggestion
This page explains the basic usage of these checks, enabling you to import all your Prometheus exposed metrics within Datadog.
```

File: content/en/integrations/guide/prometheus-host-collection.md [hunk 10]
```diff
@@ -157,11 +89,39 @@ Find below the full list of parameters that can be used for your `instances`:
 
 **Note**: All parameters but `send_distribution_buckets` and `send_distribution_counts_as_monotonic` are supported by both OpenMetrics check and Prometheus check.
 
+## Getting started
+
+### Simple metric collection
+
+Here is a simple getting started to collect metrics exposed by your Prometheus:
```
Comment (cswatt) [inline content/en/integrations/guide/prometheus-host-collection.md]:
```suggestion
To get started with collecting metrics exposed by Prometheus, follow these steps:
```

File: content/en/integrations/guide/prometheus-host-collection.md [hunk 11]
```diff
@@ -157,11 +89,39 @@ Find below the full list of parameters that can be used for your `instances`:
...
+
+1. Follow the [Prometheus Getting Started][11] documentation to start a local version of prometheus that monitors itself.
```
Comment (cswatt) [inline content/en/integrations/guide/prometheus-host-collection.md]:
```suggestion
1. Follow the [Prometheus Getting Started][11] documentation to start a local version of Prometheus that monitors itself.
```

File: content/en/integrations/guide/prometheus-host-collection.md [hunk 12]
```diff
@@ -157,11 +89,39 @@ Find below the full list of parameters that can be used for your `instances`:
...
+    init_config:
+
+    instances:
+        - prometheus_url: http://localhost:9090/metrics
+          namespace: 'documentation_example'
+          metrics:
+              - promhttp_metric_handler_requests_total: prometheus.handler.requests.total
+    ```
+
+4. [Restart the Agent][12]
```
Comment (cswatt) [inline content/en/integrations/guide/prometheus-host-collection.md]:
```suggestion
4. [Restart the Agent][12].
```

**Pull request merged**